fundamental-concepts/data-structures-stdlib/5_imports.md

```diff
@@ -0,0 +1,71 @@
+# Importing modules 1/2: Importing own modules
+
+It greatly helps to read and maintain your code, when you structure it in several modules. Let's
+import the previously created `AddressBook` module of the file `AddressBook.daml` in a new
+`Organizer` module:
+
+<pre class="file" data-filename="daml/Organizer.daml" data-target="append">
+import AddressBook
+</pre>
+
+You can now use the exported data types, functions and templates as usual:
+
+<pre class="file" data-filename="daml/Organizer.daml" data-target="append">
+data Organizer = Organizer {
+  owner: Person
+}
+</pre>
+
+Your module names can have hierarchical namespaces like `Organizer.AddressBook`. In this case your
+file needs to be in `Organizer/AddressBook.daml` relative to the root directory of your DAML source
+code and the import will look like
+
+<pre>
+import Organizer.AddressBook
+</pre>
+
+You can also be more specific on what definitions you want in scope from your import:
+
+<pre class="file" data-target="clipboard">
+import AddressBook (Person(..), Address(..))
+import Calendar (Calendar(..))
+</pre>
+
+The `(..)` means that the data type as well as all its constructors should be imported. If the
+imported module declares a name that is already declared in the importing module or any of its
+imports you can **qualify** the import:
+
+<pre class="file" data-target="clipboard">
+import qualified AddressBook as A
+import qualified Calendar as C
+</pre>
+
+Now you can use the exported names of the `AddressBook` module by prepending the qualifier
+`A.` and those of the `Calendar` module by prepending `C.`:
+
+<pre class="file" data-target="clipboard">
+data Organizer = Organizer {
+  owner: A.Person,
+  calendar: C.Calendar
+}
+</pre>
+
+The [Prelude](https://docs.daml.com/daml/stdlib/Prelude.html#module-prelude-6842) module will always
+be imported for you. You can read more on module imports and file structure
```

Comment:
```suggestion
The [Prelude](https://docs.daml.com/daml/stdlib/Prelude.html#module-prelude-6842) module will always
be imported for you along with its built-in functions, e.g., `length: [a] -> Int`. You can read more on module imports and file structure
```
Would be great to provide all the answers in the scenario. The prelude doc is pretty lengthy and noobs (like me) might not wanna read through it just to go to the next step.

Reply:
the solutions are all provided at the bottom if you really don't want to read.